test/testdata/noctx.go

```diff
@@ -0,0 +1,134 @@
+//args: -Enoctx
+package testdata
+
+import (
+	"context"
+	"net/http"
+)
+
+var newRequestPkg = http.NewRequest
+
+func Noctx() {
+	const url = "http://example.com"
+	cli := &http.Client{}
+
+	ctx := context.Background()
+	http.Get(url) // ERROR "net/http\.Get must not be called"
+	_ = http.Get  // OK
+	f := http.Get // OK
+	f(url)        // ERROR "net/http\.Get must not be called"
+
+	http.Head(url)          // ERROR "net/http\.Head must not be called"
+	http.Post(url, "", nil) // ERROR "net/http\.Post must not be called"
+	http.PostForm(url, nil) // ERROR "net/http\.PostForm must not be called"
+
+	cli.Get(url) // ERROR "\(\*net/http\.Client\)\.Get must not be called"
+	_ = cli.Get  // OK
+	m := cli.Get // OK
+	m(url)       // ERROR "\(\*net/http\.Client\)\.Get must not be called"
+
+	cli.Head(url)          // ERROR "\(\*net/http\.Client\)\.Head must not be called"
+	cli.Post(url, "", nil) // ERROR "\(\*net/http\.Client\)\.Post must not be called"
+	cli.PostForm(url, nil) // ERROR "\(\*net/http\.Client\)\.PostForm must not be called"
+
+	req, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	cli.Do(req)
+
+	req2, _ := http.NewRequestWithContext(ctx, http.MethodPost, url, nil) // OK
+	cli.Do(req2)
+
+	req3, _ := http.NewRequest(http.MethodPost, url, nil) // OK
+	req3 = req3.WithContext(ctx)
+	cli.Do(req3)
+
+	f2 := func(req *http.Request, ctx context.Context) *http.Request {
```

Comment:
What is the purpose of this function? There are few other similar introduced in this test file and I'm not sure I can grasp why.

Reply:
SSA is different for each test case. These test cases are useful for finding bugs.

If you don't need these tests for golangci-lint, I can remove them, leaving a simple test case for golangci-lint, since they are running in noctx repository.

Reply:
I'm absolutely fine with the tests - I just wanted to understand why you decided to follow such an approach :)